core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -340,10 +341,139 @@ static TEE_Result ecc_shared_secret(struct ecc_keypair *private_key,
 	return ret;
 }
 
+static TEE_Result ecc_encrypt(struct ecc_public_key *key,
+			  const uint8_t *src, size_t src_len,
+			  uint8_t *dst, size_t *dst_len)
+{
+	TEE_Result ret = TEE_ERROR_BAD_PARAMETERS;
+	struct drvcrypt_encrypt_data edata = { };
+	struct drvcrypt_ecc *ecc = NULL;
+	size_t size_bytes = 0;
+
+	if (!key || !src || !dst) {
+		CRYPTO_TRACE("Input parameters reference error!\n");
+		return ret;
+	}
+
+	key->curve = TEE_ECC_CURVE_SM2;
+	size_bytes = get_ecc_key_size_bytes(key->curve);
+	if (!size_bytes) {
+		CRYPTO_TRACE("Invalid size_bytes!\n");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	/* Verify the ciphertext length function of the key size */
+	if (*dst_len < 2 * size_bytes + src_len + TEE_SM3_HASH_SIZE) {
```

Comment:
You might want to store `2 * size_bytes + src_len + TEE_SM3_HASH_SIZE` in `output_size` variable